src/Components/Server/src/BlazorPack/BlazorPackHubProtocol.cs

```diff
@@ -0,0 +1,638 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+using MessagePack;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Internal;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.SignalR.Protocol;
+
+namespace Microsoft.AspNetCore.Components.Server.BlazorPack
+{
+    /// <summary>
+    /// Implements the SignalR Hub Protocol using MessagePack with limited type support. 
+    /// </summary>
+    public class BlazorPackHubProtocol : IHubProtocol
```

Comment:
Do we expect there will be consumers other than Blazor for this?

Reply:
Is there a super duper good reason this is public?

Reply:
Nope. Just accidental.

Reply:
That's why I asked that question. Guess I should have been more explicit.